indico/MaKaC/webinterface/rh/welcome.py

```diff
@@ -20,13 +20,21 @@
 import MaKaC.webinterface.webFactoryRegistry as webFactoryRegistry
 import MaKaC.webinterface.urlHandlers as urlHandlers
 
-class RHWelcome( base.RHDisplayBaseProtected ):
+from indico.core.db import db
+from indico.modules.users import User
+from indico.web.flask.util import url_for
+
+
+class RHWelcome(base.RHDisplayBaseProtected):
     _uh = urlHandlers.UHWelcome
 
-    def _checkParams( self, params ):
+    def _checkParams(self, params):
         self._target = conference.CategoryManager().getRoot()
 
-    def _process( self ):
-        wfReg = webFactoryRegistry.WebFactoryRegistry()
-        p = welcome.WPWelcome( self, self._target, wfReg )
-        return p.display()
+    def _process(self):
+        if not db.session.query(db.session.query(User.id).exists()).one()[0]:
```

Comment:
`find_all()` yeah, let's get 70k users in a list! :volcano:

And `User.find().count()` (or `User.query.count()` which is cleaner imo since you don't pass anything to `find()`) is SLOW. You never want to query the count of a large table unless you actually care about the count.